### ISSUE TITLE
Implement object align & distribute feature

### ISSUE THREAD

**Issue by Jose-Moreno** [synfig/synfig#1704]
**Synfig version & platform**:
1.3.16

**Issue description**:
I'm hoping the synfig community considers this, as it is honestly an essential feature for any animation software that deals with motion graphics and it is quite common in vector and motion graphics editors. 

What i'm asking here is for Synfig developers to implement the ability to allow for selective automatic distribution and alignment of multiple objects according to different combinations of rules. To me Inkscape rules are some of the most complete as seen below:

## Alignment Rules
- [ ] Align to center / origin
    - [ ] Vertical center
    - [ ] Horizontal center
- [ ] Align to top boundary
    - [ ] Inside
    - [ ] Outside
- [ ] Align to bottom boundary
    - [ ] Inside
    - [ ] Outside
- [ ] Align to left boundary
    - [ ] Inside
    - [ ] Outside
- [ ] Align to right boundary
    - [ ] Inside
    - [ ] Outside

## Distribution Rules
- [ ] Equidistant Separation
    - [ ] From Left Edge
    - [ ] From Right Edge
    - [ ] From Top Edge
    - [ ] From Bottom Edge
    - [ ] From Center
        - [ ] Vertically
        - [ ] Horizontally

- [ ] Equidistant Visual Separation (Space between objects is visually similar)
    - [ ] Vertical Gaps
    - [ ] Horizontal Gaps

## Relative Target Rules
_Note: Target should be global. Working for either alignment and distribution_
- [ ] Align relative to canvas (boundary size)
- [ ] Align relative to content (all objects bounding box)
- [ ] Align relative to selected content
- [ ] Align relative to active object (last selected)
- [ ] Align relative to active object (first selected)

**Comment by CoryAlbrecht** [comment]
There's a whole bunch of posts on the forum about a feature like this, going back almost a decade. I'm surprised it hasn't been added in that time.

**Comment by ice0** [comment]
The problem here is animation. For example, suppose you want the circle to be in the center of the rectangle. But at frame 10, you want to animate it position and move it somewhere. What should we do? Break the rules of the alignment? Do not allow animating these objects?

**Comment by Jose-Moreno** [comment]
@ice0 
> But at frame 10, you want to animate it position and move it somewhere

Then it is moved. Nothing more is expected.

The proposal here are for visual distribution tools meant for design time, not meant as animation tools directly, so you would normally align the elements on the keyframes to create the final design on the specific frame.

I personally don't expect them to be aligned during the interpolation unless I made a linear animation from start to finish where said circle has to keep.

For reference please look at Inkscape align & distribution tools https://inkscape-manuals.readthedocs.io/en/latest/align-and-distribute.html

**Comment by CoryAlbrecht** [comment]
@ice0 yeah, this isn't property to be retained animated, but rather an editing tool to help you automatically move around and place an object relative to other objects rather than dragging it manually with the mouse. It can be frustrating, for example, trying to align text used for labels manually. They're all different lengths, and all you have is that central point of the object oh, so you need to try and drag it over, drag it over a little bit more, drag it over a little bit more, progressively shorter and shorter distances, oops you went too far, try to drag it back, back a little bit more, oops you went too far again, and so on, and so forth.

All this alignment function needs to do is take the centre point and bounding box for each of the objects, and then move each of them appropriately depending on what the user selects. Left alignment, right alignment, centre alignment, and so on. It doesn't mean to retain this alignment as a property to be animated. All it's doing is changing the position just like we do was dragging the object around with a mouse.

**Comment by BSBMteam** [comment]
This is such a basic feature of most editing software, not just vector art editors.  Please add the ability to align and distribute selected objects!  You can see many people wanting this feature for years

https://forums.synfig.org/t/align-and-distribute/3318
https://forums.synfig.org/t/discussion-on-a-possible-alignment-feature/1936
